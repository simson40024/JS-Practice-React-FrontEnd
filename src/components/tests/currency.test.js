import Currency from "../header/currency";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

const InitialState = {
  currency: {
    currency: "USD",
    rate: 1,
  },
};

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const mockSelector = jest.spyOn(hooks, "useAppSelector");
const mockDispatch = jest.spyOn(hooks, "useAppDispatch");

describe("Currency", () => {
  it("should create Currency with InitialState", () => {
    mockSelector.mockImplementation((selector) => selector(InitialState));
    mockDispatch.mockReturnValue(jest.fn());
    act(() => {
      createRoot(container).render(<Currency />);
    });
    expect(container).toMatchSnapshot();
  });
});
