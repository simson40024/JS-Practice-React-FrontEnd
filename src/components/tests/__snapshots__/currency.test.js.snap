// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency should create Currency with InitialState 1`] = `
<div>
  <div
    class="currency"
  >
    <div>
      Currency
    </div>
    <select
      id="currency"
      name="Curency"
    >
      <option
        value="UAH"
      >
        UAH
      </option>
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="EUR"
      >
        EUR
      </option>
    </select>
  </div>
</div>
`;
